src/diffusers/models/molecule_gnn.py

```diff
@@ -0,0 +1,640 @@
+# Model adapted from GeoDiff https://github.com/MinkaiXu/GeoDiff
+# Model inspired by https://github.com/DeepGraphLearning/torchdrug/tree/master/torchdrug/models
+from typing import Callable, Dict, Union
+
+import numpy as np
+import torch
+import torch.nn.functional as F
+from torch import Tensor, nn
+from torch.nn import Embedding, Linear, Module, ModuleList, Sequential
+
+from torch_geometric.nn import MessagePassing, radius, radius_graph
+from torch_geometric.typing import Adj, OptPairTensor, OptTensor, Size
+from torch_geometric.utils import dense_to_sparse, to_dense_adj
+from torch_scatter import scatter_add
+from torch_sparse import SparseTensor, coalesce
+
+from ..configuration_utils import ConfigMixin, register_to_config
+from ..modeling_utils import ModelMixin
+
+
+class MultiLayerPerceptron(nn.Module):
+    """
+    Parameters:
+    Multi-layer Perceptron. Note there is no activation or dropout in the last layer.
+        input_dim (int): input dimension hidden_dim (list of int): hidden dimensions activation (str or function,
+        optional): activation function dropout (float, optional): dropout rate
+    """
+
+    def __init__(self, input_dim, hidden_dims, activation="relu", dropout=0):
+        super(MultiLayerPerceptron, self).__init__()
+
+        self.dims = [input_dim] + hidden_dims
+        if isinstance(activation, str):
+            self.activation = getattr(F, activation)
+        else:
+            self.activation = None
+        if dropout:
+            self.dropout = nn.Dropout(dropout)
+        else:
+            self.dropout = None
+
+        self.layers = nn.ModuleList()
+        for i in range(len(self.dims) - 1):
+            self.layers.append(nn.Linear(self.dims[i], self.dims[i + 1]))
+
+    def forward(self, input):
+        """"""
+        x = input
+        for i, layer in enumerate(self.layers):
+            x = layer(x)
+            if i < len(self.layers) - 1:
+                if self.activation:
+                    x = self.activation(x)
+                if self.dropout:
+                    x = self.dropout(x)
+        return x
+
+
+class ShiftedSoftplus(torch.nn.Module):
+    def __init__(self):
+        super(ShiftedSoftplus, self).__init__()
+        self.shift = torch.log(torch.tensor(2.0)).item()
+
+    def forward(self, x):
+        return F.softplus(x) - self.shift
+
+
+class CFConv(MessagePassing):
+    def __init__(self, in_channels, out_channels, num_filters, nn, cutoff, smooth):
```

Comment:
Also, if you are redefining here CFConv from the SchNet paper, I think you should init the linear/shifted softplus/linear layers (what you put in nn) here instead of passing them as arguments, as it makes it hard to follow/find the logic of the paper in your code.

Reply:
(If you want to allow something more general, the default init for this layer should still be the original one from the paper)

Reply:
Oh yeah I hadn't seen this `nn` usage until now, I really dislike that. It's very confusing (and here is clear how copy-pasted blocks of this code were).

I think this was the result of copying code from multiple files too.

Reply:
RE the second point, I don't feel as strongly about it. It was taken directly from a few codebases back. In this case, the classes `CFConv` and `InteractionBlock` could effectively become one?

Reply:
Well, if you open the SchNet paper, there is Figure 2 (I think) which describes the model arch. I think it could be interesting if you went through the figure along with your code: having CFConv and InteractionBlock separate is not a problem, but it would make CFConv easier to match with the paper it it defined the `nn` layers in directly. For further readers of this code, it will be clearer, IMO